program-runtime/src/invoke_context.rs

```diff
@@ -1199,6 +1199,7 @@ pub fn mock_process_instruction(
     transaction_accounts: Vec<TransactionAccount>,
     instruction_accounts: Vec<AccountMeta>,
     sysvar_cache_override: Option<&SysvarCache>,
+    feature_set_override: Option<Arc<FeatureSet>>,
```

Comment:
I decided to use an `Arc` here so that the caller can decide if they'd like to still use the same feature set that was passed in.

This also enables multiple calls to `mock_process_instruction()` within a single test/function to reuse the same feature set instance without cloning the whole think (just the `Arc` instead).

Not sure how big of a deal this is, and I'm guessing either way is OK, but this way was the most permissive and so that's why I've started like this.